vllm/core/scheduler.py

```diff
@@ -935,18 +993,15 @@ def _schedule_prefills(
                     waiting_queue.popleft()
                     continue
 
-            num_new_seqs = seq_group.get_max_num_running_seqs()
-            if (num_new_tokens == 0
-                    or not budget.can_schedule(num_new_tokens=num_new_tokens,
-                                               num_new_seqs=num_new_seqs)):
-                break
-
             # Can schedule this request.
             if curr_loras is not None and lora_int_id > 0:
                 curr_loras.add(lora_int_id)
             waiting_queue.popleft()
             self._allocate_and_set_running(seq_group)
 
+            if is_big:
+                self.big_prefill_requests += 1
```

Comment:
It seems improper to put `big_prefill_requests` as a scheduler attribute, because its life cycle is jus per scheduling step. If you need to pass this value to various functions when scheduling one step, you should just pass it as needed.

Reply:
As is the lifecycle is across scheduling steps, we use the values from the previous iteration for the next one. 

We could probably change it to only live per-scheduling-step though, and encapsulate all these values into a `PartialPrefillMetadata` to pass around. Would that be better?

Reply:
Updated to calculate the required metadata during each iteration instead of saving it across iterations. We didn't measure any significant performance drop from doing that